src/ElasticsearchRequestWrapper.php

```diff
@@ -0,0 +1,184 @@
+<?php
+
+namespace Drupal\elasticsearch_helper;
+
+use Drupal\elasticsearch_helper\Event\ElasticsearchEvents;
+use Drupal\elasticsearch_helper\Event\ElasticsearchOperationRequestEvent;
+use Drupal\elasticsearch_helper\Event\ElasticsearchOperationRequestResultEvent;
+use Drupal\elasticsearch_helper\Plugin\ElasticsearchIndexInterface;
+
+/**
+ * Class ElasticsearchRequestWrapper
+ */
+class ElasticsearchRequestWrapper {
```

Comment:
This perhaps need an interface to enforce these methods for classes that might extend it.